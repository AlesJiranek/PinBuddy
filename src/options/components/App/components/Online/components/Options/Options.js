import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { userLogOutAction } from 'redux-options/actions/user';
import { optionsUpdateAction, fetchOptionsAction } from 'redux-options/actions/options';
import { Paragraph, Button, Checkbox, Select } from 'theme';

const defaultViewOptions = [
  {
    value: 'all',
    text: chrome.i18n.getMessage('optionsDefaultViewAllText'),
  },
  {
    value: 'add',
    text: chrome.i18n.getMessage('optionsDefaultViewAddText'),
  },
];

class Options extends Component {
  componentDidMount() {
    this.props.fetchOptions();
  }

  render() {
    const { username } = this.props;
    const {
      defaultView,
      privateCheckboxByDefault,
      toReadChecboxByDefault,
      enableSystemNotifications,
    } = this.props.options;
    return (
      <>
        <Paragraph innerHTML t={chrome.i18n.getMessage('optionsWelcomeMessage', [username])} />
        <Button
          t={chrome.i18n.getMessage('optionsLogOut')}
          onClick={this.onCickLogOutButton}
        />

        <hr />

        <Select
          id="defaultView"
          label={chrome.i18n.getMessage('optionsDefaultView')}
          onChange={this.handleSelectChange}
          options={defaultViewOptions}
          selected={defaultView}
        />

        <Checkbox
          id="privateCheckboxByDefault"
          label={chrome.i18n.getMessage('optionsPrivateCheckboxByDefault')}
          onChange={this.handleCheckboxChange}
          checked={privateCheckboxByDefault}
        />

        <Checkbox
          id="toReadChecboxByDefault"
          label={chrome.i18n.getMessage('optionsToReadChecboxByDefault')}
          onChange={this.handleCheckboxChange}
          checked={toReadChecboxByDefault}
        />

        <Checkbox
          id="enableSystemNotifications"
          label={chrome.i18n.getMessage('optionsEnableSystemNotifications')}
          onChange={this.handleCheckboxChange}
          checked={enableSystemNotifications}
        />

        <hr />

        <ul>
          <li><code>{navigator.platform === 'MacIntel' ? '⌥ + p' : 'Alt + p'}</code> - { chrome.i18n.getMessage('optionsShortcutDescriptionToggle') }</li>
          <li><code>{navigator.platform === 'MacIntel' ? '⌥ + 1' : 'Alt + 1'}</code> - { chrome.i18n.getMessage('optionsShortcutDescriptionGoToAll') }</li>
          <li><code>{navigator.platform === 'MacIntel' ? '⌥ + 2' : 'Alt + 2'}</code> - { chrome.i18n.getMessage('optionsShortcutDescriptionGoToAdd') }</li>
          <li><code>{navigator.platform === 'MacIntel' ? '⌘ + Enter' : 'Ctrl + Enter'}</code> - { chrome.i18n.getMessage('optionsShortcutDescriptionAddBookmark') }</li>
        </ul>
      </>
    );
  }

  onCickLogOutButton = () => {
    this.props.userLogOut();
  }

  handleSelectChange = e => {
    const { id, selectedIndex } = e.target;
    this.props.optionsUpdate({
      [id]: defaultViewOptions[selectedIndex].value,
    });
  }

  handleCheckboxChange = e => {
    const { id, checked } = e.target;
    this.props.optionsUpdate({
      [id]: checked,
    });
  }

}

Options.propTypes = {
  username: PropTypes.string.isRequired,
  options: PropTypes.shape({
    defaultView: PropTypes.string.isRequired,
    privateCheckboxByDefault: PropTypes.bool.isRequired,
    toReadChecboxByDefault: PropTypes.bool.isRequired,
    enableSystemNotifications: PropTypes.bool.isRequired,
  }).isRequired,
  userLogOut: PropTypes.func.isRequired,
  fetchOptions: PropTypes.func.isRequired,
  optionsUpdate: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return {
    username: state.user.username,
    options: state.options,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    userLogOut: () => dispatch(userLogOutAction()),
    fetchOptions: () => dispatch(fetchOptionsAction()),
    optionsUpdate: option => dispatch(optionsUpdateAction(option)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Options);
