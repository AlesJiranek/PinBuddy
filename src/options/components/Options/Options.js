import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { logOut } from '../../actions/auth';
import Paragraph from '../Paragraph';
import Button from '../Button';
import Checkbox from '../Checkbox';

class Options extends Component {
  state ={
    privateCheckboxByDefault: false,
    toReadChecboxByDefault: false,
    useDescriptionMetaTag: false,
  }

  componentDidMount() {
    chrome.storage.sync.get(options => {
      this.setState({
        ...options,
      });
    });
  }

  render() {
    const { username } = this.props;
    return (
      <>
        <Paragraph innerHTML t={chrome.i18n.getMessage('optionsWelcomeMessage', [username])} />
        <Button
          t={chrome.i18n.getMessage('optionsSignOut')}
          onClick={this.onCickLogOutButton}
        />

        <hr />

        <Checkbox
          id="privateCheckboxByDefault"
          label={chrome.i18n.getMessage('optionsPrivateCheckboxByDefault')}
          onChange={this.handleCheckboxChange}
          checked={this.state.privateCheckboxByDefault}
        />

        <Checkbox
          id="toReadChecboxByDefault"
          label={chrome.i18n.getMessage('optionsToReadChecboxByDefault')}
          onChange={this.handleCheckboxChange}
          checked={this.state.toReadChecboxByDefault}
        />

        <Checkbox
          id="useDescriptionMetaTag"
          label={chrome.i18n.getMessage('optionsUseDescriptionMetaTag')}
          onChange={this.handleCheckboxChange}
          checked={this.state.useDescriptionMetaTag}
        />

      </>
    );
  }

  onCickLogOutButton = () => {
    chrome.storage.local.clear(() => this.props.logOut());
  }

  handleCheckboxChange = e => {
    const { id } = e.target;
    this.setState(state => {
      return {
        [id]: !state[id],
      };
    }, () => {
      chrome.storage.sync.set({ ...this.state });
    });
  }
}

Options.propTypes = {
  username: PropTypes.string.isRequired,
  logOut: PropTypes.func.isRequired,
};

const mapStateToProps = (state, ownProps) => {
  return {
    username: state.auth.username,
  };
};

const mapDispatchToProps = dispatch => {
  return {
    logOut: () => dispatch(logOut()),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Options);
