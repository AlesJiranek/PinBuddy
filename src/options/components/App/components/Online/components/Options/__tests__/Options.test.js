import React from 'react';
import { shallow } from 'enzyme';

import { Options } from '../Options';

describe('<Options />', () => {

  const defaultProps = {
    username: 'pinbuddy',
    options: {
      defaultView: 'all',
      privateCheckboxByDefault: false,
      toReadChecboxByDefault: false,
      enableSystemNotifications: false,
    },
    userLogOut: jest.fn(),
    fetchOptions: jest.fn(),
    optionsUpdate: jest.fn(),
  };

  it('should render correctly', () => {
    const tree = shallow(
      <Options
        {...defaultProps}
      />
    );
    expect(tree).toMatchSnapshot();
  });

});
