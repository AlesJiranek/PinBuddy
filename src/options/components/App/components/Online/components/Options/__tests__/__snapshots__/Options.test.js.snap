// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Options /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Options
    fetchOptions={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    options={
      Object {
        "defaultView": "all",
        "enableSystemNotifications": false,
        "privateCheckboxByDefault": false,
        "toReadChecboxByDefault": false,
      }
    }
    optionsUpdate={[MockFunction]}
    userLogOut={[MockFunction]}
    username="pinbuddy"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Paragraph
          innerHTML={true}
          t="optionsWelcomeMessage"
        />,
        <Button
          disabled={false}
          onClick={[Function]}
          secondary={false}
          t="optionsLogOut"
          title={null}
          type="button"
        />,
        <hr />,
        <Select
          id="defaultView"
          label="optionsDefaultView"
          onChange={[Function]}
          options={
            Array [
              Object {
                "text": "optionsDefaultViewAllText",
                "value": "all",
              },
              Object {
                "text": "optionsDefaultViewAddText",
                "value": "add",
              },
            ]
          }
          selected="all"
        />,
        <Checkbox
          checked={false}
          id="privateCheckboxByDefault"
          label="optionsPrivateCheckboxByDefault"
          onChange={[Function]}
        />,
        <Checkbox
          checked={false}
          id="toReadChecboxByDefault"
          label="optionsToReadChecboxByDefault"
          onChange={[Function]}
        />,
        <Checkbox
          checked={false}
          id="enableSystemNotifications"
          label="optionsEnableSystemNotifications"
          onChange={[Function]}
        />,
        <hr />,
        <ul>
          <li>
            <code>
              Alt + p
            </code>
             - 
            optionsShortcutDescriptionToggle
          </li>
          <li>
            <code>
              Alt + 1
            </code>
             - 
            optionsShortcutDescriptionGoToAll
          </li>
          <li>
            <code>
              Alt + 2
            </code>
             - 
            optionsShortcutDescriptionGoToAdd
          </li>
          <li>
            <code>
              Ctrl + Enter
            </code>
             - 
            optionsShortcutDescriptionAddBookmark
          </li>
          <li>
            <code>
              Ctrl + Enter
            </code>
             - 
            optionsShortcutDescriptionOpenInBackground
          </li>
          <li>
            <code>
              Ctrl + Backspace
            </code>
             - 
            optionsShortcutDescriptionDeleteBookmark
          </li>
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "innerHTML": true,
          "t": "optionsWelcomeMessage",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": false,
          "onClick": [Function],
          "secondary": false,
          "t": "optionsLogOut",
          "title": null,
          "type": "button",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "id": "defaultView",
          "label": "optionsDefaultView",
          "onChange": [Function],
          "options": Array [
            Object {
              "text": "optionsDefaultViewAllText",
              "value": "all",
            },
            Object {
              "text": "optionsDefaultViewAddText",
              "value": "add",
            },
          ],
          "selected": "all",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checked": false,
          "id": "privateCheckboxByDefault",
          "label": "optionsPrivateCheckboxByDefault",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checked": false,
          "id": "toReadChecboxByDefault",
          "label": "optionsToReadChecboxByDefault",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "checked": false,
          "id": "enableSystemNotifications",
          "label": "optionsEnableSystemNotifications",
          "onChange": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <code>
                Alt + p
              </code>
               - 
              optionsShortcutDescriptionToggle
            </li>,
            <li>
              <code>
                Alt + 1
              </code>
               - 
              optionsShortcutDescriptionGoToAll
            </li>,
            <li>
              <code>
                Alt + 2
              </code>
               - 
              optionsShortcutDescriptionGoToAdd
            </li>,
            <li>
              <code>
                Ctrl + Enter
              </code>
               - 
              optionsShortcutDescriptionAddBookmark
            </li>,
            <li>
              <code>
                Ctrl + Enter
              </code>
               - 
              optionsShortcutDescriptionOpenInBackground
            </li>,
            <li>
              <code>
                Ctrl + Backspace
              </code>
               - 
              optionsShortcutDescriptionDeleteBookmark
            </li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <code>
                  Alt + p
                </code>,
                " - ",
                "optionsShortcutDescriptionToggle",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Alt + p",
                },
                "ref": null,
                "rendered": "Alt + p",
                "type": "code",
              },
              " - ",
              "optionsShortcutDescriptionToggle",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <code>
                  Alt + 1
                </code>,
                " - ",
                "optionsShortcutDescriptionGoToAll",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Alt + 1",
                },
                "ref": null,
                "rendered": "Alt + 1",
                "type": "code",
              },
              " - ",
              "optionsShortcutDescriptionGoToAll",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <code>
                  Alt + 2
                </code>,
                " - ",
                "optionsShortcutDescriptionGoToAdd",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Alt + 2",
                },
                "ref": null,
                "rendered": "Alt + 2",
                "type": "code",
              },
              " - ",
              "optionsShortcutDescriptionGoToAdd",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <code>
                  Ctrl + Enter
                </code>,
                " - ",
                "optionsShortcutDescriptionAddBookmark",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Ctrl + Enter",
                },
                "ref": null,
                "rendered": "Ctrl + Enter",
                "type": "code",
              },
              " - ",
              "optionsShortcutDescriptionAddBookmark",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <code>
                  Ctrl + Enter
                </code>,
                " - ",
                "optionsShortcutDescriptionOpenInBackground",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Ctrl + Enter",
                },
                "ref": null,
                "rendered": "Ctrl + Enter",
                "type": "code",
              },
              " - ",
              "optionsShortcutDescriptionOpenInBackground",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <code>
                  Ctrl + Backspace
                </code>,
                " - ",
                "optionsShortcutDescriptionDeleteBookmark",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Ctrl + Backspace",
                },
                "ref": null,
                "rendered": "Ctrl + Backspace",
                "type": "code",
              },
              " - ",
              "optionsShortcutDescriptionDeleteBookmark",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Paragraph
            innerHTML={true}
            t="optionsWelcomeMessage"
          />,
          <Button
            disabled={false}
            onClick={[Function]}
            secondary={false}
            t="optionsLogOut"
            title={null}
            type="button"
          />,
          <hr />,
          <Select
            id="defaultView"
            label="optionsDefaultView"
            onChange={[Function]}
            options={
              Array [
                Object {
                  "text": "optionsDefaultViewAllText",
                  "value": "all",
                },
                Object {
                  "text": "optionsDefaultViewAddText",
                  "value": "add",
                },
              ]
            }
            selected="all"
          />,
          <Checkbox
            checked={false}
            id="privateCheckboxByDefault"
            label="optionsPrivateCheckboxByDefault"
            onChange={[Function]}
          />,
          <Checkbox
            checked={false}
            id="toReadChecboxByDefault"
            label="optionsToReadChecboxByDefault"
            onChange={[Function]}
          />,
          <Checkbox
            checked={false}
            id="enableSystemNotifications"
            label="optionsEnableSystemNotifications"
            onChange={[Function]}
          />,
          <hr />,
          <ul>
            <li>
              <code>
                Alt + p
              </code>
               - 
              optionsShortcutDescriptionToggle
            </li>
            <li>
              <code>
                Alt + 1
              </code>
               - 
              optionsShortcutDescriptionGoToAll
            </li>
            <li>
              <code>
                Alt + 2
              </code>
               - 
              optionsShortcutDescriptionGoToAdd
            </li>
            <li>
              <code>
                Ctrl + Enter
              </code>
               - 
              optionsShortcutDescriptionAddBookmark
            </li>
            <li>
              <code>
                Ctrl + Enter
              </code>
               - 
              optionsShortcutDescriptionOpenInBackground
            </li>
            <li>
              <code>
                Ctrl + Backspace
              </code>
               - 
              optionsShortcutDescriptionDeleteBookmark
            </li>
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "innerHTML": true,
            "t": "optionsWelcomeMessage",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "onClick": [Function],
            "secondary": false,
            "t": "optionsLogOut",
            "title": null,
            "type": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "id": "defaultView",
            "label": "optionsDefaultView",
            "onChange": [Function],
            "options": Array [
              Object {
                "text": "optionsDefaultViewAllText",
                "value": "all",
              },
              Object {
                "text": "optionsDefaultViewAddText",
                "value": "add",
              },
            ],
            "selected": "all",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": false,
            "id": "privateCheckboxByDefault",
            "label": "optionsPrivateCheckboxByDefault",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": false,
            "id": "toReadChecboxByDefault",
            "label": "optionsToReadChecboxByDefault",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "checked": false,
            "id": "enableSystemNotifications",
            "label": "optionsEnableSystemNotifications",
            "onChange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                <code>
                  Alt + p
                </code>
                 - 
                optionsShortcutDescriptionToggle
              </li>,
              <li>
                <code>
                  Alt + 1
                </code>
                 - 
                optionsShortcutDescriptionGoToAll
              </li>,
              <li>
                <code>
                  Alt + 2
                </code>
                 - 
                optionsShortcutDescriptionGoToAdd
              </li>,
              <li>
                <code>
                  Ctrl + Enter
                </code>
                 - 
                optionsShortcutDescriptionAddBookmark
              </li>,
              <li>
                <code>
                  Ctrl + Enter
                </code>
                 - 
                optionsShortcutDescriptionOpenInBackground
              </li>,
              <li>
                <code>
                  Ctrl + Backspace
                </code>
                 - 
                optionsShortcutDescriptionDeleteBookmark
              </li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <code>
                    Alt + p
                  </code>,
                  " - ",
                  "optionsShortcutDescriptionToggle",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Alt + p",
                  },
                  "ref": null,
                  "rendered": "Alt + p",
                  "type": "code",
                },
                " - ",
                "optionsShortcutDescriptionToggle",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <code>
                    Alt + 1
                  </code>,
                  " - ",
                  "optionsShortcutDescriptionGoToAll",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Alt + 1",
                  },
                  "ref": null,
                  "rendered": "Alt + 1",
                  "type": "code",
                },
                " - ",
                "optionsShortcutDescriptionGoToAll",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <code>
                    Alt + 2
                  </code>,
                  " - ",
                  "optionsShortcutDescriptionGoToAdd",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Alt + 2",
                  },
                  "ref": null,
                  "rendered": "Alt + 2",
                  "type": "code",
                },
                " - ",
                "optionsShortcutDescriptionGoToAdd",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <code>
                    Ctrl + Enter
                  </code>,
                  " - ",
                  "optionsShortcutDescriptionAddBookmark",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Ctrl + Enter",
                  },
                  "ref": null,
                  "rendered": "Ctrl + Enter",
                  "type": "code",
                },
                " - ",
                "optionsShortcutDescriptionAddBookmark",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <code>
                    Ctrl + Enter
                  </code>,
                  " - ",
                  "optionsShortcutDescriptionOpenInBackground",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Ctrl + Enter",
                  },
                  "ref": null,
                  "rendered": "Ctrl + Enter",
                  "type": "code",
                },
                " - ",
                "optionsShortcutDescriptionOpenInBackground",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <code>
                    Ctrl + Backspace
                  </code>,
                  " - ",
                  "optionsShortcutDescriptionDeleteBookmark",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Ctrl + Backspace",
                  },
                  "ref": null,
                  "rendered": "Ctrl + Backspace",
                  "type": "code",
                },
                " - ",
                "optionsShortcutDescriptionDeleteBookmark",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
